### PR TITLE
No naked argument Exceptions

### PR DESCRIPTION
**Pull request by paulvanbrenk**
Update argumentexceptions to include meaningful information.

### CODE REVIEW

File: src/EditorFeatures/Core/Extensibility/SignatureHelp/SignatureHelpItems.cs [hunk 1]
```diff
@@ -64,12 +64,12 @@ internal class SignatureHelpItems
 
             if (argumentIndex < 0)
             {
-                throw new ArgumentException();
+                throw new ArgumentException("argumentIndex < 0", "argumentIndex");
```
Comment (tmeschter) [inline src/EditorFeatures/Core/Extensibility/SignatureHelp/SignatureHelpItems.cs]: Use `nameof(argumentIndex)`?

File: src/EditorFeatures/Core/Extensibility/SignatureHelp/SignatureHelpItems.cs [hunk 2]
```diff
@@ -64,12 +64,12 @@ internal class SignatureHelpItems
...
             }
 
             if (argumentCount < argumentIndex)
             {
-                throw new ArgumentException();
+                throw new ArgumentException("argumentCount < argumentIndex", "argumentIndex");
```
Comment (tmeschter) [inline src/EditorFeatures/Core/Extensibility/SignatureHelp/SignatureHelpItems.cs]: Same here.

**Comment by jaredpar** [comment]
:+1:  pending @tmeschter feedback

**Comment by paulvanbrenk** [comment]
I like the fancy.

**Comment by tmeschter** [comment]
@dotnet-bot Retest this, please.

**Comment by dpoeschl** [comment]
retest this please

**Comment by dpoeschl** [comment]
retest this please

**Comment by VladimirReshetnikov** [comment]
:+1:

**Pull request merged**